tests/Feature/ApplicantControllerTest.php

```diff
@@ -0,0 +1,140 @@
+        $response->assertJsonFragment([
+            'citizenship_declaration_id' => $applicant->citizenship_declaration_id,
+            'veteran_status_id' => $applicant->veteran_status_id,
+            'applicant_classifications' => $applicant->applicant_classifications->toArray(),
+        ]);
+    }
+
+    public function testUpdateProfile(): void
+    {
+        $applicant = factory(Applicant::class)->create();
+        $applicant->applicant_classifications()->delete();
+
+        // Ensure that submitting applicant classifications creates new elements.
+        $newApplicantClassification = [
+            'id' => 12345,
+            'applicant_id' => $applicant->id,
+            'classification_id' => Classification::inRandomOrder()->first()->id,
+            'level' => 5,
+            'order' => 1,
+        ];
+        $updateData = [
+            'citizenship_declaration_id' => CitizenshipDeclaration::inRandomOrder()->first()->id,
+            'veteran_status_id' => VeteranStatus::inRandomOrder()->first()->id,
+            'applicant_classifications' => [
+                0 => $newApplicantClassification
+            ],
+        ];
+        $response = $this->actingAs($applicant->user)
+            ->putJson("$this->baseUrl/applicant/$applicant->id/profile", $updateData);
+        $response->assertOk();
+
+        $expectedData = [
+            'citizenship_declaration_id' => $updateData['citizenship_declaration_id'],
+            'veteran_status_id' => $updateData['veteran_status_id'],
+            'applicant_classifications' => $applicant->applicant_classifications->toArray(),
+        ];
+        $response->assertJsonFragment($expectedData);
+        $this->assertDatabaseHas(
+            'applicant_classifications',
+            $applicant->applicant_classifications->first()->toArray()
+        );
+
+        // Ensure that submitting applicant classifications deletes elements that were not resubmitted
+        // and creates new elements/updates old elements.
+        $oldApplicantClassification = $applicant->applicant_classifications->first()->toArray();
+        $updateData = [
+            'citizenship_declaration_id' => CitizenshipDeclaration::inRandomOrder()->first()->id,
+            'veteran_status_id' => VeteranStatus::inRandomOrder()->first()->id,
+            'applicant_classifications' => [
+                0 => $oldApplicantClassification,
+                1 => $newApplicantClassification,
+            ],
+        ];
+        $response = $this->actingAs($applicant->user)
+            ->putJson("$this->baseUrl/applicant/$applicant->id/profile", $updateData);
+        $response->assertOk();
+
+        $applicant->refresh();
+        $expectedData = [
+            'citizenship_declaration_id' => $updateData['citizenship_declaration_id'],
+            'veteran_status_id' => $updateData['veteran_status_id'],
+            'applicant_classifications' => $applicant->applicant_classifications->toArray(),
```

Comment:
Here (and a few lines later in the for loop) data is taken from `$applicant` and used in assertions. That proves that the response is correct, but it doesn't prove that `$applicant` was updated in the way we want. I think we should take the data from `$updateData` instead, like for the other fields.